packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts

```diff
@@ -219,36 +243,64 @@ export class StatefulDagLMDGHOST implements ILMDGHOST {
     if (this.bestJustifiedCheckpoint && (!this.justified ||
       this.bestJustifiedCheckpoint.epoch > this.justified.epoch)) {
       this.setJustified(this.bestJustifiedCheckpoint);
+      this.ensureCorrectBestTargets();
     }
   }
 
   public addBlock(slot: Slot, blockRootBuf: Uint8Array, parentRootBuf: Uint8Array,
-    justifiedCheckpoint?: Checkpoint, finalizedCheckpoint?: Checkpoint): void {
+    justifiedCheckpoint: Checkpoint, finalizedCheckpoint: Checkpoint): void {
     this.synced = false;
     const blockRoot = toHexString(blockRootBuf);
     const parentRoot = toHexString(parentRootBuf);
+    const justifiedEpoch = justifiedCheckpoint.epoch;
+    const finalizedEpoch = finalizedCheckpoint.epoch;
     // ensure blockRoot exists
     const node: Node = this.nodes[blockRoot] || new Node({
       slot,
       blockRoot,
       parent: this.nodes[parentRoot],
+      justifiedEpoch,
+      finalizedEpoch,
     });
     // best target is the node itself
     node.bestTarget = node;
     this.nodes[blockRoot] = node;
 
     // if parent root exists, link to blockRoot
     if (this.nodes[parentRoot]) {
-      this.nodes[parentRoot].addChild(node);
+      this.nodes[parentRoot].addChild(node, justifiedEpoch, finalizedEpoch);
```

Comment:
I think we're supposed to be using the store's justified / finalized information, eg: `this.justified` and `this.finalized`.
Also, it would be good to check against the whole checkpoint (epoch + root), rather than just the epoch.

Reply:
we want to track the added block's checkpoints in the node, not the store's checkpoints. Later on we can compare block's checkpoints against the current store's checkpoints. This is similar to `head_state = store.block_states[block_root]` in the spec.

yeah I'll make the check against `epoch + root` instead of just `epoch`.

Reply:
ah seems like I even don't need to change `addChild` interface as justified/finalized is already in the child